pkg/xds/generator/egress/external_services_generator.go

```diff
@@ -146,14 +145,11 @@ func (*ExternalServicesGenerator) generateCDS(
 
 func (*ExternalServicesGenerator) buildServices(
 	endpointMap core_xds.EndpointMap,
-	localZone string,
-	meshResources *core_xds.MeshResources,
 ) map[string]bool {
 	services := map[string]bool{}
 
 	for serviceName, endpoints := range endpointMap {
-		if len(endpoints) > 0 && endpoints[0].IsExternalService() &&
-			(!meshResources.Mesh.ZoneEgressEnabled() || endpoints[0].IsReachableFromZone(localZone)) {
```

Comment:
This part
```
!meshResources.Mesh.ZoneEgressEnabled() || endpoints[0].IsReachableFromZone(localZone)
```
was
1) wrong, because it should be `meshResources.Mesh.ZoneEgressEnabled()`
but also
2) unnecessary, because endpoints in endpoint map are prefiltered with `endpoints[0].IsReachableFromZone(localZone)` anyways, so this part was always true for external services. We build this endpoint map per single egress.